app/definition/ApiDefinitionFactory.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package definition
+
+import definition.APIStatus.APIStatus
+import javax.inject.{Inject, Singleton}
+import play.api.Logger
+import config.{AppConfig, FeatureSwitch}
+import Versions._
+
+@Singleton
+class ApiDefinitionFactory @Inject()(appConfig: AppConfig) {
+
+  private val readScope = "read:self-assessment"
```

Comment:
If this is a template, would it be better to have values like this as `???` rather than actual Strings? Otherwise they might get overlooked.

Reply:
Actually that might make the tests blow up. Maybe not. Worth a try though?

Reply:
Missed this one. Have set few other places with api-specific strings with text enclosed in hashes to catch the eye. Maybe ??? would be better. Not sure if it matters that tests blow up - as it forces implementations to be added.

Can then add to readme that wherever ??? appears should be fixed up. (WDYT?)

Reply:
Yeah sounds good 👍🏼